### ISSUE TITLE
'Enter to Chat' can be nonsensical

### ISSUE THREAD

**Issue by Jalopyy** [rsbmatt/OSBuddy#372]
I prefer ETC because I played WoW before recently coming to OSRS, but certain things about the function don't work in a logical way.

1) If quickreplying to someone (Tab), you have to press enter to enable chat _before_ typing anything, despite the whisper dialog box superseding the chat entirely. 
2) Same thing happens with the input box for searching for things on the GE. 

This all probably happens in some other places where you have to type in something and the box overtakes the normal chat window (adding friends, ignoring, renaming, etc). It would be much more sound that if a box like that appears, ETC is disabled so you can just type normally. Since the message that tells you to hit Enter to talk won't always reappear unless you hit Esc, sometimes it can be annoying to talk to someone in whispers without a lot of leaving the window, making sure enter is hit, hitting tab again...etc.

tl;dr w/Enter to Chat if a dialog input box appears over the main chat, disable ETC or make it so it automatically enables the use of key presses normally.

**Comment by rsbfox** [comment]
Just confirming you know you can use middle mouse or alt key + left drag to move camera?

I don't like the ETC plugin. And I think it's weird it's on by default.

**Comment by Jalopyy** [comment]
Yea I know, nothing about this ticket has to do with camera movement. It's mostly just that you have to press enter to type in dialog boxes which one would only ever open if they meant to type anything anyway, it feels inconsistent design wise.

**Comment by Jalopyy** [comment]
Also the reason I care about ETC is because a decent amount of the time I have to play on a laptop w/o a mouse. So I prefer WSAD in those cases.

**Comment by rsbmatt** [comment]
I agree. This plugin needs a revamp, which will probably be happening soon. I'll try and implement changes to account for the issues you brought up.
